client/packages/common/src/ui/components/steppers/HorizontalStepper/HorizontalStepper.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { FC } from 'react';
+import Box from '@mui/material/Box';
+import Stepper from '@mui/material/Stepper';
+import Step from '@mui/material/Step';
+import StepLabel from '@mui/material/StepLabel';
+import Typography from '@mui/material/Typography';
+import StepConnector, {
+  stepConnectorClasses,
+} from '@mui/material/StepConnector';
+import {
+  GlobalStyles,
+  PaletteColor,
+  StepIconProps,
+  useTheme,
+} from '@mui/material';
+
+export interface StepDefinition {
+  active?: boolean;
+  completed?: boolean;
+  error?: boolean;
+  icon?: React.ReactNode;
+  label: string;
+  optional?: React.ReactNode;
+}
+
+export type StepperColour = 'primary' | 'secondary';
+
+interface StepperProps {
+  steps: StepDefinition[];
+  colour?: StepperColour;
+}
+
+type PaletteColour = Omit<PaletteColor, 'contrastText'> & { error: string };
+
+const getConnector = (paletteColour: PaletteColour) => () => {
+  const style = {
+    '&.MuiStepConnector-root': {
+      left: 'calc(-50% + 15px)',
+      right: 'calc(50% + 15px)',
+    },
+    [`& .${stepConnectorClasses.line}`]: {
+      borderColor: paletteColour.light,
+      // The width is for the connector which is significantly thicker than the default.
+      borderWidth: '4px 0 0 0',
+    },
+    [`&.${stepConnectorClasses.active}`]: {
+      [`& .${stepConnectorClasses.line}`]: {
+        borderColor: paletteColour.main,
+      },
+    },
+    [`&.${stepConnectorClasses.completed}`]: {
+      [`& .${stepConnectorClasses.line}`]: {
+        borderColor: paletteColour.main,
+      },
+    },
+  };
+
+  return <StepConnector sx={style} />;
+};
+
+const getCircle =
+  (paletteColour: PaletteColour) =>
+  ({ active, completed, icon, error }: StepIconProps) => {
+    const colors = {
+      borderColor: paletteColour.light,
+      backgroundColor: '#fff',
+      color: paletteColour.main,
+    };
+
+    if (completed) {
+      colors.backgroundColor = paletteColour.main;
+      colors.borderColor = paletteColour.main;
+      colors.color = '#fff';
+    }
+
+    if (active) {
+      colors.backgroundColor = '#fff';
+      colors.color = paletteColour.dark;
+    }
+
+    if (!!error) {
+      colors.color = paletteColour.error;
+    }
+
+    const style = {
+      border: 'solid',
+      borderWidth: '3px',
+      width: '30px',
+      height: '30px',
+      borderRadius: '30px',
+      textAlign: 'center' as 'center' | 'left' | 'right',
```

Comment:
hmm, wish TS was a bit better at type inference (i am already getting used to rust). 
Could also put all of those inside the div instead of creating `style` variable (although I kind of like breaking things down into variable rather then doing things inline)

Reply:
yeah - this sort of thing is particularly annoying.
Often when inline, ts is better at figuring out the style types. I've moved this one inline and it looks pretty good, and have taken out the `textAlign` typing and that's working fine.